src/api/app/models/package.rb

```diff
@@ -547,6 +547,10 @@ def is_of_kind?(kind)
     package_kinds.where(kind: kind).exists?
   end
 
+  def ignored_requests
```

Comment:
I find an ignored_requests in the package model should be about requests not a yaml config file

Reply:
the ignored_requests are really a property of a (staging) project not of a package.

How about a StagingProjectDecorator ?

Reply:
For now I just wanted to get this out of the controller and simplify the code that handles this file. And as it is right now this is stored in a package file.
So I think it's not too strange to have the method here.

We are going to sit together this sprint and think about how and where we can move the model code for staging project and distribution.

In the long run I would say that tracked requests should now be stored in a project description and the dashboard package is replaced by some sane configuration as well.

Reply:
But why move it at all? And if so, why to a pretty random model? there is nothing active recordy about ignored_requests.

Reply:
Right now this data is stored in a file, so it's stored in the backend. The next closest active record would be the package model.

As said I want to wrap this code in a method and simplify it.

Reply:
Good intentions - but why not leave that method in the controller?

Reply:
Why not moving it where it, technically, belongs to?

Reply:
because you're just adding the 118th function to a file that has 1523 lines already. And one that is only used in one controller in one action. That is what rails developers refer to as obese models.

And later if you decide that ignored_requests is no longer a thing in the staging controller and we rather - let's say display requests tagged with emojis - you will forget about that one function in the package model reading a file. That's why codeclimate refers to the maintainability of the Package model as rating F. Beacuse in 118 functions it's hard to figure out what each does.

And the fact that the ignored_requests are stored in a yaml file does not mean they technically belong to a Package. A file belongs to a package and a package belongs to a project - so 'technically' you can put everything everywhere. The right split is the art of programming :)

Reply:
But I am not adding 118 functions, I am adding 1. And a very slick one as well.

> And the fact that the ignored_requests are stored in a yaml file does not mean they technically belong to a Package. A file belongs to a package and a package belongs to a project - so 'technically' you can put everything everywhere. The right split is the art of programming :)

Thanks for teaching me:-) Still, source files are fetched via package objects and the ignored requests are currently fetched via a source file.
It seems we both agree that this is a bad design decision. But it's how it is right now.

(I think I already wrote that to me the next step would be making the dashboard a thing. And then we can convert the existing files and project descriptions)